src/ussd_nurse.js

```diff
@@ -1,21 +1,154 @@
+            );
+
+            sbm = new StageBasedMessaging(
+                new JsonApi(self.im, {}),
+                self.im.config.services.stage_based_messaging.token,
+                self.im.config.services.stage_based_messaging.url
+            );
+        };
+
+        // override normal state adding
+        self.add = function(name, creator) {
+            self.states.add(name, function(name, opts) {
+                // if (!interrupt || !self.timed_out(self.im))
+                    return creator(name, opts);
+
+                // interrupt = false;
+                // var timeout_opts = opts || {};
+                // timeout_opts.name = name;
+                // return self.states.create('st_timed_out', timeout_opts);
+            });
+        };
 
+        self.has_active_nurseconnect_subscription = function(id) {
+            return sbm
+            .list_active_subscriptions(id)
+            .then(function(active_subs_response) {
+                var active_subs = active_subs_response.results;
+                for (var i=0; i < active_subs.length; i++) {
+                    // get the subscription messageset
+                    return sbm
+                    .get_messageset(active_subs[i].messageset)
+                    .then(function(messageset) {
+                        if (messageset.short_name.indexOf('nurseconnect') > -1) {
+                            return true;
+                        }
+                    });
+                }
+                return false;
+            });
         };
 
-        self.states.add("state_start", function(name) {
-            return new EndState(name, {
-                text: $("Welcome to The Department of Health's ussd_nurse.js"),
+    // START STATE
+
+        self.add('state_start', function(name) {
+            var msisdn = utils.normalize_msisdn(self.im.user.addr, '27');
+            return is
+            .get_or_create_identity({"msisdn": msisdn})
+            .then(function(identity) {
+                if (identity.details.nurseconnect) {
+                    if ((!_.isUndefined(identity.details.nurseconnect))
+                        && (identity.details.nurseconnect.working_on !== "")) {
+                        self.im.user.set_answer("user", identity);
+                        return is
+                        .get_or_create_identity({"msisdn": identity.details.nurseconnect.working_on})
+                        .then(function(working_on_identity) {
+                            self.im.user.set_answer("contact", working_on_identity);
+
+                            return self.states.create('state_route');
+                        });
+                    }
+                } else {
+                    self.im.user.set_answer("contact", identity);
+                }
+
+                self.im.user.set_answer("user", identity);
+
+                return self.states.create('state_route');
+            });
+        });
```

Comment:
This state was a bit of a misunderstanding. Please combine state_start and state_route into one state.  It should do:
1. Clear the user answers (similar to our other apps)
2. `get_or_create_identity` for the "user" (as you've done in line 77)
3. `self.im.user.set_answer("user", identity)` (as you've done in line 82)
We can do the "contact" saving when we get to it in the code (i.e. when they choose to register someone else).
Does that make sense?